### ISSUE TITLE
False positive on NP_NONNULL_PARAM_VIOLATION (?)

### ISSUE THREAD

**Issue by SoerenHenning** [spotbugs/spotbugs#1923]
Hi, I'm calling the very simple [Apache Beam `KV`](https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/values/KV.java) static method:

```java
  public static <K extends @Nullable Object, V extends @Nullable Object> KV<K, V> of(
      K key, V value) {
    return new KV<>(key, value);
  }
```

As far as I see, it should be totally fine for parameters `key` and `value` to be `null`.

However, for the following code

```java
final KV<String, Object> kv = KV.of("abc", null);
```

I'm getting a

```
NP Bug type NP_NONNULL_PARAM_VIOLATION
Null passed for non-null parameter of org.apache.beam.sdk.values.KV.of(Object, Object) in <...>
```

To me this seems wrong so is that a bug in Spotbugs or am I misunderstanding something?

I'm using the following setup:

* **Gradle version:** 7.3.3
* **Spotbugs Gradle plugin version:** 5.0.4
* **Spotbugs Tool version:** 4.5.3

**Comment by gtoison** [comment]
This should be fixed by #2502
